### PR TITLE
Same site session cookie

### LINKED ISSUE #10787: Update the <httpSession/> server configuration for SameSite
- Proposed: `<httpSession cookieSameSite="None" cookieSecure="true"/>` 
cookieSecure is an existing metaType so no action is necessary there.

- Ensure the metaType uses an Enum. `Lax,None,Strict,Disabled` 

- Determine default value or if we just don't set SameSite by default like we do today.

- WebContainer to provide a way for the SameSite attribute value to be passed to them from where the addCookie call is made on the Response object in #10820

- Any additional logging that needs to be updated etc for logging the SameSite attribute?

### LINKED ISSUE #10922: Update the com.ibm.ws.session component to leverage the cookieSameSite metaType
1) Leverage the `<httpSession cookieSameSite=lax/>` metaType in `com.ibm.ws.session`. Add logic specified in #10787
2) Enable the enable/add the session related tests.

### PR DESCRIPTION
**Pull request by pnicolucci**
fixes #10922

### CODE REVIEW

**Comment by pnicolucci** [comment]
#build

**Comment by pnicolucci** [comment]
#run-libby-bot

File: dev/com.ibm.ws.session/resources/OSGI-INF/l10n/metatype.properties [hunk 1]
```diff
@@ -51,6 +51,14 @@ cookie.secure.desc=Specifies that the session cookies include the secure field.
 cookie.http.only=Include HttpOnly field in session cookies
 cookie.http.only.desc=Specifies that session cookies include the HttpOnly field. Browsers that support the HttpOnly field do not enable cookies to be accessed by client-side scripts. Using the HttpOnly field will help prevent cross-site scripting attacks.
 
+# the word SameSite should not be translated
+cookie.same.site=Include SameSite field in session cookies
+cookie.same.site.desc=Specifies which SameSite attribute value should be used for session cookies.
```
Comment (helyarp) [inline dev/com.ibm.ws.session/resources/OSGI-INF/l10n/metatype.properties]: Change to "Specifies the SameSite attribute value to use for session cookies."

File: dev/com.ibm.ws.session/resources/OSGI-INF/l10n/metatype.properties [hunk 2]
```diff
@@ -51,6 +51,14 @@ cookie.secure.desc=Specifies that the session cookies include the secure field.
...
+cookie.same.site.disabled=Do not set a SameSite field on the session cookies
```
Comment (helyarp) [inline dev/com.ibm.ws.session/resources/OSGI-INF/l10n/metatype.properties]: Do you mean "field" or "attribute value"?

File: dev/com.ibm.ws.session/resources/OSGI-INF/l10n/metatype.properties [hunk 3]
```diff
@@ -51,6 +51,14 @@ cookie.secure.desc=Specifies that the session cookies include the secure field.
...
+cookie.same.site.lax=Set the session cookies SameSite field to Lax
```
Comment (helyarp) [inline dev/com.ibm.ws.session/resources/OSGI-INF/l10n/metatype.properties]: It'd look better with  no "s" at the end of "cookies." That is, use "session cookie SameSite field."

Comment (helyarp) [reply dev/com.ibm.ws.session/resources/OSGI-INF/l10n/metatype.properties]: This comment applies to the two following strings as well.

**Comment by pnicolucci** [comment]
@helyarp thanks for the review, I've pushed up my latest changes for your final review. Let me know if you have any questions or comments.

**Comment by helyarp** [comment]
@pnicolucci Suggestion: Add a period to the end of full sentences. Thx for the quick changes. I've added the ID reviewed label.

**Comment by skasund** [comment]
L2 message review not required

**Comment by pnicolucci** [comment]
#build

**Comment by pnicolucci** [comment]
#build

**Comment by pnicolucci** [comment]
#run-libby-bot

**Pull request merged**